tests/cpu_bugs/xen_mitigations.pm

```diff
@@ -0,0 +1,349 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: CPU BUGS on Linux kernel check
+# Maintainer: James Wang <jnwang@suse.com>
+#
+# This testsuite for the spectre control mitigations on XEN hypervisor.
+#
+# From xen's oneline doc:
+# http://xenbits.xen.org/docs/4.12-testing/misc/xen-command-line.html#xpti-x86
+# http://xenbits.xen.org/docs/4.12-testing/misc/xen-command-line.html#spec-ctrl-x86
+# http://xenbits.xen.org/docs/4.12-testing/misc/xen-command-line.html#pv-l1tf-x86
+#
+# We testing each of parameter. To check the influence of each of them.
+#
+# There is a data structure to store the data that for compare with the data in runtime system.
+# When we found something of unexpection string, we call them go die.
+# When we didn't find something of expection string, we call them go die too.
+#
+# mitigations_list = {
+#	'<parameter name>' => {
+#		<the value of parameter> => {
+#			<secnario1> => {
+#				#determine string.
+#				determine => {'<cmd>' => ['']},
+#				#expection string. If it doesn't appear go die
+#				expection => {'<cmd>' => ['ecpection string1','expection string2']},
+#				#unexpection string. If it appears go die.
+#				unexpection => {'<cmd>' => ['unexpection string1']}
+#			}
+#			<secnario2> => {
+#				.....
+#			}
+#		}
+#	}
+# }
+# Put into xpti, spec-ctrl, into this structure. and The testing code will read all them and
+# one by one execution checking.
+#
+
+package xen_mitigations;
+use strict;
+use warnings;
+use base "consoletest";
+use bootloader_setup;
+use Mitigation;
+use ipmi_backend_utils;
+use power_action_utils 'power_action';
+use testapi;
+use utils;
+
+my $mitigations_list =
+  {
+    #command-line parameter name
+    xpti => {
+        #command-line the value of parameter.
+        #It point to a array that include TWO elements.
+
+        "true" => {
+            "default" => {
+                expection => {'xl dmesg' => ['XPTI (64-bit PV only): Dom0 enabled, DomU enabled']},
+                unexpection => {'xl dmesg' => ['XPTI (64-bit PV only): Dom0 enabled, DomU disabled', 'XPTI (64-bit PV only): Dom0 disabled, DomU enabled', 'XPTI (64-bit PV only): Dom0 disabled, DomU disabled']}
+            }
+        },
+        "false" => {
+            "default" => {
+                expection => {'xl dmesg' => ['XPTI (64-bit PV only): Dom0 disabled, DomU disabled']},
```

Comment:
`expection/unexpection` - not English words as far as I know . Did you mean `expectation` ? Since you can not say `unexpectation` , I would suggest you two options : 
1. `expected/unexpected` 
2. replace hash with array and just define that 0 element of array - will contain unexpected and 1st element - will contain expected

Reply:
2. is my old way, but be changed by @Amrysliu ;-). Ahaha.

1. language problem. I will fix it.